eudico/fxmodules/libp2p.go

```diff
@@ -0,0 +1,110 @@
+package fxmodules
+
+import (
+	"github.com/filecoin-project/lotus/node/config"
+	"github.com/filecoin-project/lotus/node/modules"
+	"github.com/filecoin-project/lotus/node/modules/lp2p"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	ci "github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"go.uber.org/fx"
+	"time"
+)
+
+func Libp2p(cfg *config.Common) fx.Option {
+	// private providers
+	privateProviders := fx.Options(
+		fx.Provide(
+			fx.Private,
+
+			// Host settings
+			lp2p.DefaultTransports,
+			lp2p.SmuxTransport(),
+			lp2p.NoRelay(),
+			lp2p.Security(true, false),
+
+			// Routing
+			modules.RecordValidator,
+
+			// Services
+			lp2p.AutoNATService,
+
+			// Services (connection management)
```

Comment:
We should double-check that the defaults of the Libp2p config are the ones set here (just in case). If this is the case feel free to remove this.

Reply:
This is always being overridden by https://github.com/consensus-shipyard/lotus/blob/83414fb6f1eed326c4aa5f389afa2e80e0ebb8b5/node/builder.go#L290